Drop IsDeprecatedWeakRefSmartPointerException for PlatformGamepad

cdumez opened the pull request:
#### 37b3abb46e9257cb5c087723e7b435f272deb483
<pre>
Drop IsDeprecatedWeakRefSmartPointerException for PlatformGamepad
<a href="https://bugs.webkit.org/show_bug.cgi?id=304230">https://bugs.webkit.org/show_bug.cgi?id=304230</a>

Reviewed by Darin Adler and Adrian Perez de Castro.

* Source/WebCore/Modules/gamepad/GamepadManager.cpp:
(WebCore::GamepadManager::platformGamepadConnected):
(WebCore::GamepadManager::platformGamepadInputActivity):
* Source/WebCore/Modules/gamepad/NavigatorGamepad.cpp:
(WebCore::NavigatorGamepad::gamepads):
(WebCore::NavigatorGamepad::gamepadsBecameVisible):
* Source/WebCore/Modules/webxr/WebXRGamepad.cpp:
* Source/WebCore/Modules/webxr/WebXRGamepad.h:
* Source/WebCore/Modules/webxr/WebXRInputSource.cpp:
(WebCore::WebXRInputSource::WebXRInputSource):
(WebCore::WebXRInputSource::update):
* Source/WebCore/platform/gamepad/PlatformGamepad.h:
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepad.h:
(WebCore::GameControllerGamepad::ensureProtectedHapticEngines): Deleted.
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepad.mm:
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepadProvider.mm:
(WebCore::GameControllerGamepadProvider::controllerDidConnect):
(WebCore::GameControllerGamepadProvider::controllerDidDisconnect):
(WebCore::GameControllerGamepadProvider::makeInvisibleGamepadsVisible):
* Source/WebCore/platform/gamepad/libwpe/GamepadLibWPE.cpp:
* Source/WebCore/platform/gamepad/libwpe/GamepadLibWPE.h:
* Source/WebCore/platform/gamepad/mac/Dualshock3HIDGamepad.h:
* Source/WebCore/platform/gamepad/mac/GenericHIDGamepad.cpp:
* Source/WebCore/platform/gamepad/mac/GenericHIDGamepad.h:
* Source/WebCore/platform/gamepad/mac/HIDGamepad.cpp:
* Source/WebCore/platform/gamepad/mac/HIDGamepad.h:
* Source/WebCore/platform/gamepad/mac/HIDGamepadProvider.mm:
(WebCore::HIDGamepadProvider::deviceAdded):
(WebCore::HIDGamepadProvider::valuesChanged):
* Source/WebCore/platform/gamepad/mac/LogitechGamepad.h:
* Source/WebCore/platform/gamepad/mac/MultiGamepadProvider.h:
(WebCore::MultiGamepadProvider::PlatformGamepadWrapper::PlatformGamepadWrapper): Deleted.
* Source/WebCore/platform/gamepad/mac/MultiGamepadProvider.mm:
(WebCore::MultiGamepadProvider::platformGamepadConnected):
* Source/WebCore/platform/gamepad/mac/StadiaHIDGamepad.h:
* Source/WebCore/platform/gamepad/manette/ManetteGamepad.cpp:
* Source/WebCore/platform/gamepad/manette/ManetteGamepad.h:
* Source/WebCore/testing/MockGamepad.cpp:
* Source/WebCore/testing/MockGamepad.h:
(WebCore::MockGamepad::wasConnected): Deleted.
* Source/WebCore/testing/MockGamepadProvider.cpp:
(WebCore::MockGamepadProvider::setMockGamepadDetails):
(WebCore::MockGamepadProvider::connectMockGamepad):
(WebCore::MockGamepadProvider::disconnectMockGamepad):
(WebCore::MockGamepadProvider::setMockGamepadAxisValue):
(WebCore::MockGamepadProvider::setMockGamepadButtonValue):
* Source/WebKit/UIProcess/Gamepad/UIGamepadProvider.cpp:
(WebKit::UIGamepadProvider::platformGamepadInputActivity):
* Source/WebKit/UIProcess/Gamepad/wpe/PlatformGamepadWPE.cpp:
* Source/WebKit/UIProcess/Gamepad/wpe/PlatformGamepadWPE.h:
* Source/WebKit/WebProcess/Gamepad/WebGamepad.cpp:
* Source/WebKit/WebProcess/Gamepad/WebGamepad.h:
* Source/WebKit/WebProcess/Gamepad/WebGamepadProvider.cpp:
(WebKit::WebGamepadProvider::gamepadConnected):
(WebKit::WebGamepadProvider::gamepadActivity):

Canonical link: <a href="https://commits.webkit.org/304574@main">https://commits.webkit.org/304574@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8492ca11f09c56e9b4af5c743af39568bf886e96

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/136017 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/8369 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/47291 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/143716 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/88990 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/137886 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/9038 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/8216 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/103939 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/88990 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e9fbe508-1c6e-43c0-8742-0968eb79cbfb) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/138963 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/6554 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/121887 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/84816 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/6246 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/3880 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/4318 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/115515 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/40079 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/146470 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/8053 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/40646 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/112295 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/8071 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/6756 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/112688 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28589 "Built successfully and passed tests") | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/164/builds/6144 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/118188 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/62032 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/8102 "Built successfully") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/36258 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/7821 "Failed to checkout and rebase branch from PR 55461") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/71659 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/8042 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/152/builds/7899 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

aperezdc reviewed (approved):
LGTM, for the cross-port changes and the WPE/GTK bits. The Cocoa part looks reasonable, but I am definitely not an expert there.

darinadler commented on Source/WebCore/platform/gamepad/cocoa/GameControllerGamepadProvider.mm:
```diff
@@ -141,22 +141,23 @@ static bool shouldExcludeGameController(GCController *)
     LOG(Gamepad, "GameControllerGamepadProvider controller %p removed", controller);
 
-    auto removedGamepad = m_gamepadMap.take((__bridge CFTypeRef)controller);
+    std::unique_ptr<GameControllerGamepad> removedGamepad = m_gamepadMap.take((__bridge CFTypeRef)controller);
```
Does this work?
```suggestion
    std::unique_ptr removedGamepad = m_gamepadMap.take((__bridge CFTypeRef)controller);
```

cdumez replied:
I did not try but I know that `auto` gave a safer cpp static analysis false positive. When this happens, @rniwa said to use the explicit type, which worked (safer cpp bot is now green).

I am not sure if `std::unique_ptr` would satisfy the static analyzer but I can try.

cdumez replied:
Looks like safer cpp is happy with `std::unique_ptr`.

cdumez committed:
Drop IsDeprecatedWeakRefSmartPointerException for PlatformGamepad
https://bugs.webkit.org/show_bug.cgi?id=304230

Reviewed by Darin Adler and Adrian Perez de Castro.

* Source/WebCore/Modules/gamepad/GamepadManager.cpp:
(WebCore::GamepadManager::platformGamepadConnected):
(WebCore::GamepadManager::platformGamepadInputActivity):
* Source/WebCore/Modules/gamepad/NavigatorGamepad.cpp:
(WebCore::NavigatorGamepad::gamepads):
(WebCore::NavigatorGamepad::gamepadsBecameVisible):
* Source/WebCore/Modules/webxr/WebXRGamepad.cpp:
* Source/WebCore/Modules/webxr/WebXRGamepad.h:
* Source/WebCore/Modules/webxr/WebXRInputSource.cpp:
(WebCore::WebXRInputSource::WebXRInputSource):
(WebCore::WebXRInputSource::update):
* Source/WebCore/platform/gamepad/PlatformGamepad.h:
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepad.h:
(WebCore::GameControllerGamepad::ensureProtectedHapticEngines): Deleted.
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepad.mm:
* Source/WebCore/platform/gamepad/cocoa/GameControllerGamepadProvider.mm:
(WebCore::GameControllerGamepadProvider::controllerDidConnect):
(WebCore::GameControllerGamepadProvider::controllerDidDisconnect):
(WebCore::GameControllerGamepadProvider::makeInvisibleGamepadsVisible):
* Source/WebCore/platform/gamepad/libwpe/GamepadLibWPE.cpp:
* Source/WebCore/platform/gamepad/libwpe/GamepadLibWPE.h:
* Source/WebCore/platform/gamepad/mac/Dualshock3HIDGamepad.h:
* Source/WebCore/platform/gamepad/mac/GenericHIDGamepad.cpp:
* Source/WebCore/platform/gamepad/mac/GenericHIDGamepad.h:
* Source/WebCore/platform/gamepad/mac/HIDGamepad.cpp:
* Source/WebCore/platform/gamepad/mac/HIDGamepad.h:
* Source/WebCore/platform/gamepad/mac/HIDGamepadProvider.mm:
(WebCore::HIDGamepadProvider::deviceAdded):
(WebCore::HIDGamepadProvider::valuesChanged):
* Source/WebCore/platform/gamepad/mac/LogitechGamepad.h:
* Source/WebCore/platform/gamepad/mac/MultiGamepadProvider.h:
(WebCore::MultiGamepadProvider::PlatformGamepadWrapper::PlatformGamepadWrapper): Deleted.
* Source/WebCore/platform/gamepad/mac/MultiGamepadProvider.mm:
(WebCore::MultiGamepadProvider::platformGamepadConnected):
* Source/WebCore/platform/gamepad/mac/StadiaHIDGamepad.h:
* Source/WebCore/platform/gamepad/manette/ManetteGamepad.cpp:
* Source/WebCore/platform/gamepad/manette/ManetteGamepad.h:
* Source/WebCore/testing/MockGamepad.cpp:
* Source/WebCore/testing/MockGamepad.h:
(WebCore::MockGamepad::wasConnected): Deleted.
* Source/WebCore/testing/MockGamepadProvider.cpp:
(WebCore::MockGamepadProvider::setMockGamepadDetails):
(WebCore::MockGamepadProvider::connectMockGamepad):
(WebCore::MockGamepadProvider::disconnectMockGamepad):
(WebCore::MockGamepadProvider::setMockGamepadAxisValue):
(WebCore::MockGamepadProvider::setMockGamepadButtonValue):
* Source/WebKit/UIProcess/Gamepad/UIGamepadProvider.cpp:
(WebKit::UIGamepadProvider::platformGamepadInputActivity):
* Source/WebKit/UIProcess/Gamepad/wpe/PlatformGamepadWPE.cpp:
* Source/WebKit/UIProcess/Gamepad/wpe/PlatformGamepadWPE.h:
* Source/WebKit/WebProcess/Gamepad/WebGamepad.cpp:
* Source/WebKit/WebProcess/Gamepad/WebGamepad.h:
* Source/WebKit/WebProcess/Gamepad/WebGamepadProvider.cpp:
(WebKit::WebGamepadProvider::gamepadConnected):
(WebKit::WebGamepadProvider::gamepadActivity):

Canonical link: https://commits.webkit.org/304574@main
